src/test/java/org/jabref/logic/pdf/search/retrieval/PdfSearcherTest.java

```diff
@@ -0,0 +1,99 @@
+package org.jabref.logic.pdf.search.retrieval;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+
+import org.jabref.logic.pdf.search.indexing.PdfIndexer;
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.model.pdf.search.PdfSearchResults;
+import org.jabref.preferences.FilePreferences;
+
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PdfSearcherTest {
+
+    private PdfSearcher search;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        FilePreferences filePreferences = mock(FilePreferences.class);
+        // given
+        BibDatabase database = new BibDatabase();
+        BibDatabaseContext context = mock(BibDatabaseContext.class);
+        when(context.getFileDirectories(Mockito.any())).thenReturn(Collections.singletonList(Path.of("src/test/resources/pdfs")));
+        when(context.getFulltextIndexPath()).thenReturn(Path.of("src/test/resources/luceneTestIndex"));
```

Comment:
Better is to use the @TempDir , this ensure that anything inside is deleted after the test
https://www.baeldung.com/junit-5-temporary-directory